langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/DefaultMistralAiHelper.java

```diff
@@ -0,0 +1,120 @@
+        }
+        return value;
+    }
+
+    public static String formattedURLForRetrofit(String baseUrl) {
+        return baseUrl.endsWith("/") ? baseUrl : baseUrl + "/";
+    }
+
+    public static List<MistralChatMessage> toMistralAiMessages(List<ChatMessage> messages) {
+        return messages.stream()
+                .map(DefaultMistralAiHelper::toMistralAiMessage)
+                .collect(toList());
+    }
+
+    public static MistralChatMessage toMistralAiMessage(ChatMessage message) {
+        return MistralChatMessage.builder()
+                .role(toMistralAiRole(message.type()))
+                .content(message.text())
+                .build();
+    }
+
+    private static MistralRoleName toMistralAiRole(ChatMessageType chatMessageType) {
+        switch (chatMessageType) {
+            case SYSTEM:
+                return MistralRoleName.SYSTEM;
+            case  AI:
+                return MistralRoleName.ASSISTANT;
+            case USER:
+                return MistralRoleName.USER;
+            default:
+                throw new IllegalArgumentException("Unknown chat message type: " + chatMessageType);
+        }
+
+    }
+
+    static TokenUsage tokenUsageFrom(MistralUsageInfo mistralAiUsage) {
+        if (mistralAiUsage == null) {
+            return null;
+        }
+        return new TokenUsage(
+                mistralAiUsage.getPromptTokens(),
+                mistralAiUsage.getCompletionTokens(),
+                mistralAiUsage.getTotalTokens()
+        );
+    }
+
+    static FinishReason finishReasonFrom(String mistralAiFinishReason) {
+        if (mistralAiFinishReason == null) {
+            return null;
+        }
+        switch (mistralAiFinishReason) {
+            case "stop":
+                return STOP;
+            case "length":
+                return LENGTH;
+            case "model_length":
+            default:
+                return null;
+        }
+    }
+
+    static String getHeaders(Headers headers){
+       return StreamSupport.stream(headers.spliterator(),false).map(header -> {
+           String headerKey = header.component1();
+           String headerValue = header.component2();
+           if (headerKey.equals("Authorization")) {
+               headerValue = maskAuthorizationHeaderValue(headerValue);
+           }
+           return  String.format("[%s: %s]", headerKey, headerValue);
+       }).collect(Collectors.joining(", "));
+    }
+
+    private static String maskAuthorizationHeaderValue(String authorizationHeaderValue) {
+        try {
+            Matcher matcher = MISTRAI_API_KEY_BEARER_PATTERN.matcher(authorizationHeaderValue);
+            StringBuffer sb = new StringBuffer();
+
+            while (matcher.find()) {
+                String bearer = matcher.group(1);
+                String token = matcher.group(2);
+                matcher.appendReplacement(sb, bearer + " " + token.substring(0, 7) + "..." + token.substring(token.length() - 7));
```

Comment:
```suggestion
                matcher.appendReplacement(sb, bearer + " " + token.substring(0, 2) + "..." + token.substring(token.length() - 2));
```
Just to be on the safe side. 4 symbols is enough to identify one of your keys...